src/iso/fs.ts

```diff
@@ -127,13 +135,33 @@ export class IsoFS extends Readonly(Sync(FileSystem)) {
 
 		for (const part of path.split('/').slice(1)) {
 			if (!dir.isDirectory()) return;
-			dir = dir.directory.get(part);
+			const directory: Directory = dir.directory;
+			let next: DirectoryRecord | undefined = directory.get(part);
+			if (!next && this.options.caseFold) {
+				const foldedPart = this._caseFold(part);
+				for (const [name, record] of directory) {
+					if (this._caseFold(name) === foldedPart) {
+						next = record;
+						break;
+					}
+				}
+			}
+
+			dir = next;
 			if (!dir) return;
 		}
 
 		return dir;
 	}
 
+	private _caseFold(original: string): string {
+		if (!this.options.caseFold) {
+			return original;
+		}
+
+		return this.options.caseFold === 'upper' ? original.toUpperCase() : original.toLowerCase();
+	}
```

Comment:
This is duplicated with ZipFS as well. To reduce duplcation, I'm thinking:
- Make `options` `public` but add `@internal`
- Change case fold to be `@internal`, and take the FS: `_caseFold(fs: FileSystem, path: string) => string`
- Both FSes use the shared case fold function